app/controllers/sessions_controller.rb

```diff
@@ -51,8 +53,10 @@ def mfa_create
     @user = User.find(session[:mfa_user])
     session.delete(:mfa_user)
 
-    if @user&.mfa_enabled? && @user&.otp_verified?(params[:otp])
+    if @user&.mfa_enabled? && @user&.otp_verified?(params[:otp]) && !session_expired?
```

Comment:
This expression is getting complex, it might be time to extract a variable or method here.

Reply:
I might also suggest calling this `session_active?` and reversing the polarity so that you don't need to negate it in this boolean expression.

Reply:
Addressed these in latest commits.